components/clp-package-utils/clp_package_utils/scripts/native/archive_manager.py

```diff
@@ -0,0 +1,355 @@
+            if end_ts is not None:
+                query += " AND end_timestamp <= %s"
+                params = params + (end_ts,)
+
+            db_cursor.execute(query, params)
+            results = db_cursor.fetchall()
+
+            archive_ids = [result["id"] for result in results]
+            if 0 == len(archive_ids):
+                logger.info("No archives found within specified time range.")
+                return 0
+
+            logger.info(f"Found {len(archive_ids)} archives within the specified time range.")
+            for archive_id in archive_ids:
+                logger.info(archive_id)
+                archive_path = archives_dir / archive_id
+                if not archive_path.is_dir():
+                    logger.warning(f"Archive {archive_id} in database not found on disk.")
+
+    except Exception:
+        logger.exception("Failed to find archives from the database.")
+        return -1
+
+    logger.info(f"Finished finding archives from the database.")
+    return 0
+
+
+def _delete_archives(
+    archives_dir: Path,
+    database_config: Database,
+    delete_handler: DeleteHandler,
+    dry_run: bool = False,
+) -> int:
+    """
+    Deletes archives from both metadata database and disk based on provided SQL query.
+    :param archives_dir:
+    :param database_config:
+    :param delete_handler: Object to handle differences between by-filter and by-ids delete types.
+    :param dry_run: If True, no changes will be made to the database or disk.
+    :return: 0 on success, -1 otherwise.
+    """
+
+    archive_ids: List[str]
+    logger.info("Starting to delete archives from the database.")
+    try:
+        sql_adapter = SQL_Adapter(database_config)
+        clp_db_connection_params = database_config.get_clp_connection_params_and_type(True)
+        table_prefix = clp_db_connection_params["table_prefix"]
+        with closing(sql_adapter.create_connection(True)) as db_conn, closing(
+            db_conn.cursor(dictionary=True)
+        ) as db_cursor:
+            if dry_run:
+                logger.info("Running in dry-run mode.")
+
+            criteria = delete_handler.get_criteria()
+            params = delete_handler.get_params()
+
+            db_cursor.execute(
+                f"""
+                DELETE FROM `{table_prefix}archives`
+                WHERE {criteria}
+                RETURNING id
+                """,
+                params,
+            )
+            results = db_cursor.fetchall()
+
+            if 0 == len(results):
+                logger.info(delete_handler.get_not_found_message)
+                return 0
+
+            archive_ids = [result["id"] for result in results]
+            delete_handler.validate_results(archive_ids)
+
+            ids_string = ", ".join(f"'{archive_id}'" for archive_id in archive_ids)
+
+            db_cursor.execute(
+                f"""
+                DELETE FROM `{table_prefix}files`
+                WHERE archive_id in ({ids_string})
```

Comment:
With the change on line 309, I believe that we can remove ()
```suggestion
                WHERE archive_id in {ids_string}
```